rootfs/webapp/acarshub_db.py

```diff
@@ -201,32 +247,50 @@ def add_message_from_json(message_type, message_from_json):
         else:
             print(f"[database] Unidenitied key: {index}")
 
-    if os.getenv("DB_SAVEALL", default=False) or text is not None or libacars is not None or \
-       dsta is not None or depa is not None or eta is not None or gtout is not None or \
-       gtin is not None or wloff is not None or wlin is not None or lat is not None or \
-       lon is not None or alt is not None:
-        # create a session for this thread to write
+    try:
         session = db_session()
-        # write the message
-        if os.getenv("DEBUG_LOGGING", default=False):
-            print("[database] writing to the database")
-            print(f"[database] writing message: {message_from_json}")
+        count = session.query(messagesCount).first()
+        count.total += 1
+
+        if error is not None and error > 0:
+            count.errors += 1
+        else:
+            count.good += 1
+
+        found_freq = session.query(messagesFreq).filter(messagesFreq.freq == f"{freq}" and messagesFreq.freq_type == message_type).first()
+
+        if found_freq is not None:
+            found_freq.count += 1
+        else:
+            session.add(messagesFreq(freq=f"{freq}", freq_type=message_type, count=1))
+
+        if os.getenv("DB_SAVEALL", default=False) or text is not None or libacars is not None or \
```

Comment:
Wow that's quite the if statement :)

Reply:
I wish I could write it in a simpler way, because it is...